spec/rspec/core/world_spec.rb

```diff
@@ -21,6 +21,33 @@ module RSpec::Core
         world.register(example_group)
         expect(world.example_groups).to include(example_group)
       end
+
+      it "invokes before_register callbacks" do
+        world.before_register do |example_group|
+          examples     = example_group.examples
+          new_examples = []
+          examples.each do |ex|
+            caps = [{:browserName => :firefox}, {:browserName => :chrome}]
+            caps.each_with_index do |cap, index|
+              ex_with_cap = ex.clone
+              new_id      = "#{ex_with_cap.id}_#{index}_#{cap}"
+              ex_with_cap.instance_variable_set(:@id, new_id)
+              new_examples << ex_with_cap
+            end
+          end
+
+          example_group.instance_variable_set(:@examples, new_examples)
```

Comment:
Messing with the instance variables of the example group is definitely not something we'll support.  As such, I'd prefer the spec for this (which should show an example of the usage of this feature) not demonstrate it by doing something that we don't support users doing.